Epick_d no longer compile on recent Debian testing

https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.9-Ic-141/NewKernel_d/TestReport_lrineau_Debian-Testing.gz
> /usr/include/eigen3/Eigen/src/Core/util/XprHelper.h:74:8: error: no type named 'Literal' in 'struct Eigen::NumTraits<CGAL::Interval_nt<false> >'

--- a/Number_types/include/CGAL/CORE_BigFloat.h
+++ b/Number_types/include/CGAL/CORE_BigFloat.h
@@ -531,6 +531,7 @@ namespace Eigen {
     typedef CORE::BigFloat Real;
     typedef CORE::BigFloat NonInteger;
     typedef CORE::BigFloat Nested;
+    typedef CORE::BigFloat Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/CORE_BigFloat.h
+++ b/Number_types/include/CGAL/CORE_BigFloat.h
@@ -526,7 +526,9 @@ template <> class Real_embeddable_traits< CORE::BigFloat >
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::BigFloat>
+   : GenericNumTraits<CORE::BigFloat>
   {
     typedef CORE::BigFloat Real;
     typedef CORE::BigFloat NonInteger;

--- a/Number_types/include/CGAL/CORE_BigFloat.h
+++ b/Number_types/include/CGAL/CORE_BigFloat.h
@@ -526,9 +526,7 @@ template <> class Real_embeddable_traits< CORE::BigFloat >
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::BigFloat>
-   : GenericNumTraits<CORE::BigFloat>
   {
     typedef CORE::BigFloat Real;
     typedef CORE::BigFloat NonInteger;

--- a/Number_types/include/CGAL/CORE_BigInt.h
+++ b/Number_types/include/CGAL/CORE_BigInt.h
@@ -207,6 +207,7 @@ namespace Eigen {
     typedef CORE::BigInt Real;
     typedef CORE::BigRat NonInteger;
     typedef CORE::BigInt Nested;
+    typedef CORE::BigInt Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/CORE_BigInt.h
+++ b/Number_types/include/CGAL/CORE_BigInt.h
@@ -202,9 +202,7 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::BigInt>
-   : GenericNumTraits<CORE::BigInt>
   {
     typedef CORE::BigInt Real;
     typedef CORE::BigRat NonInteger;

--- a/Number_types/include/CGAL/CORE_BigInt.h
+++ b/Number_types/include/CGAL/CORE_BigInt.h
@@ -202,7 +202,9 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::BigInt>
+   : GenericNumTraits<CORE::BigInt>
   {
     typedef CORE::BigInt Real;
     typedef CORE::BigRat NonInteger;

--- a/Number_types/include/CGAL/CORE_BigRat.h
+++ b/Number_types/include/CGAL/CORE_BigRat.h
@@ -241,6 +241,7 @@ namespace Eigen {
     typedef CORE::BigRat Real;
     typedef CORE::BigRat NonInteger;
     typedef CORE::BigRat Nested;
+    typedef CORE::BigRat Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/CORE_BigRat.h
+++ b/Number_types/include/CGAL/CORE_BigRat.h
@@ -236,7 +236,9 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::BigRat>
+   : GenericNumTraits<CORE::BigRat>
   {
     typedef CORE::BigRat Real;
     typedef CORE::BigRat NonInteger;

--- a/Number_types/include/CGAL/CORE_BigRat.h
+++ b/Number_types/include/CGAL/CORE_BigRat.h
@@ -236,9 +236,7 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::BigRat>
-   : GenericNumTraits<CORE::BigRat>
   {
     typedef CORE::BigRat Real;
     typedef CORE::BigRat NonInteger;

--- a/Number_types/include/CGAL/CORE_Expr.h
+++ b/Number_types/include/CGAL/CORE_Expr.h
@@ -196,6 +196,7 @@ namespace Eigen {
     typedef CORE::Expr Real;
     typedef CORE::Expr NonInteger;
     typedef CORE::Expr Nested;
+    typedef CORE::Expr Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/CORE_Expr.h
+++ b/Number_types/include/CGAL/CORE_Expr.h
@@ -191,9 +191,7 @@ template <> class Real_embeddable_traits< CORE::Expr >
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::Expr>
-   : GenericNumTraits<CORE::Expr>
   {
     typedef CORE::Expr Real;
     typedef CORE::Expr NonInteger;

--- a/Number_types/include/CGAL/CORE_Expr.h
+++ b/Number_types/include/CGAL/CORE_Expr.h
@@ -191,7 +191,9 @@ template <> class Real_embeddable_traits< CORE::Expr >
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CORE::Expr>
+   : GenericNumTraits<CORE::Expr>
   {
     typedef CORE::Expr Real;
     typedef CORE::Expr NonInteger;

--- a/Number_types/include/CGAL/Gmpfi.h
+++ b/Number_types/include/CGAL/Gmpfi.h
@@ -342,6 +342,7 @@ namespace Eigen {
     typedef CGAL::Gmpfi Real;
     typedef CGAL::Gmpfi NonInteger;
     typedef CGAL::Gmpfi Nested;
+    typedef CGAL::Gmpfi Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/Gmpfi.h
+++ b/Number_types/include/CGAL/Gmpfi.h
@@ -337,7 +337,9 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpfi>
+   : GenericNumTraits<CGAL::Gmpfi>
   {
     typedef CGAL::Gmpfi Real;
     typedef CGAL::Gmpfi NonInteger;

--- a/Number_types/include/CGAL/Gmpfi.h
+++ b/Number_types/include/CGAL/Gmpfi.h
@@ -337,9 +337,7 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpfi>
-   : GenericNumTraits<CGAL::Gmpfi>
   {
     typedef CGAL::Gmpfi Real;
     typedef CGAL::Gmpfi NonInteger;

--- a/Number_types/include/CGAL/Gmpfr.h
+++ b/Number_types/include/CGAL/Gmpfr.h
@@ -162,6 +162,7 @@ namespace Eigen {
     typedef CGAL::Gmpfr Real;
     typedef CGAL::Gmpfr NonInteger;
     typedef CGAL::Gmpfr Nested;
+    typedef CGAL::Gmpfr Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/Gmpfr.h
+++ b/Number_types/include/CGAL/Gmpfr.h
@@ -157,9 +157,7 @@ public INTERN_RET::Real_embeddable_traits_base<Gmpfr,CGAL::Tag_true>{
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpfr>
-   : GenericNumTraits<CGAL::Gmpfr>
   {
     typedef CGAL::Gmpfr Real;
     typedef CGAL::Gmpfr NonInteger;

--- a/Number_types/include/CGAL/Gmpfr.h
+++ b/Number_types/include/CGAL/Gmpfr.h
@@ -157,7 +157,9 @@ public INTERN_RET::Real_embeddable_traits_base<Gmpfr,CGAL::Tag_true>{
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpfr>
+   : GenericNumTraits<CGAL::Gmpfr>
   {
     typedef CGAL::Gmpfr Real;
     typedef CGAL::Gmpfr NonInteger;

--- a/Number_types/include/CGAL/Gmpq.h
+++ b/Number_types/include/CGAL/Gmpq.h
@@ -141,6 +141,7 @@ namespace Eigen {
     typedef CGAL::Gmpq Real;
     typedef CGAL::Gmpq NonInteger;
     typedef CGAL::Gmpq Nested;
+    typedef CGAL::Gmpq Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/Gmpq.h
+++ b/Number_types/include/CGAL/Gmpq.h
@@ -136,7 +136,9 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpq>
+   : GenericNumTraits<CGAL::Gmpq>
   {
     typedef CGAL::Gmpq Real;
     typedef CGAL::Gmpq NonInteger;

--- a/Number_types/include/CGAL/Gmpq.h
+++ b/Number_types/include/CGAL/Gmpq.h
@@ -136,9 +136,7 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpq>
-   : GenericNumTraits<CGAL::Gmpq>
   {
     typedef CGAL::Gmpq Real;
     typedef CGAL::Gmpq NonInteger;

--- a/Number_types/include/CGAL/Gmpz.h
+++ b/Number_types/include/CGAL/Gmpz.h
@@ -215,6 +215,7 @@ namespace Eigen {
     typedef CGAL::Gmpz Real;
     typedef CGAL::Gmpq NonInteger;
     typedef CGAL::Gmpz Nested;
+    typedef CGAL::Gmpz Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/Gmpz.h
+++ b/Number_types/include/CGAL/Gmpz.h
@@ -210,7 +210,9 @@ class Modular_traits< Gmpz > {
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpz>
+   : GenericNumTraits<CGAL::Gmpz>
   {     
     typedef CGAL::Gmpz Real;
     typedef CGAL::Gmpq NonInteger;

--- a/Number_types/include/CGAL/Gmpz.h
+++ b/Number_types/include/CGAL/Gmpz.h
@@ -210,9 +210,7 @@ class Modular_traits< Gmpz > {
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Gmpz>
-   : GenericNumTraits<CGAL::Gmpz>
   {     
     typedef CGAL::Gmpz Real;
     typedef CGAL::Gmpq NonInteger;

--- a/Number_types/include/CGAL/Interval_nt.h
+++ b/Number_types/include/CGAL/Interval_nt.h
@@ -1263,9 +1263,7 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<bool b> struct NumTraits<CGAL::Interval_nt<b> >
-    : GenericNumTraits<CGAL::Interval_nt<b> >
   {
     typedef CGAL::Interval_nt<b> Real;
     typedef CGAL::Interval_nt<b> NonInteger;

--- a/Number_types/include/CGAL/Interval_nt.h
+++ b/Number_types/include/CGAL/Interval_nt.h
@@ -1270,6 +1270,7 @@ namespace Eigen {
     typedef CGAL::Interval_nt<b> Real;
     typedef CGAL::Interval_nt<b> NonInteger;
     typedef CGAL::Interval_nt<b> Nested;
+    typedef double Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/Interval_nt.h
+++ b/Number_types/include/CGAL/Interval_nt.h
@@ -1263,7 +1263,9 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<bool b> struct NumTraits<CGAL::Interval_nt<b> >
+    : GenericNumTraits<CGAL::Interval_nt<b> >
   {
     typedef CGAL::Interval_nt<b> Real;
     typedef CGAL::Interval_nt<b> NonInteger;

--- a/Number_types/include/CGAL/Lazy_exact_nt.h
+++ b/Number_types/include/CGAL/Lazy_exact_nt.h
@@ -1428,6 +1428,7 @@ namespace Eigen {
     // typedef CGAL::Lazy_exact_nt<ET> NonInteger;
     typedef CGAL::Lazy_exact_nt<typename NumTraits<ET>::NonInteger> NonInteger;
     typedef CGAL::Lazy_exact_nt<ET> Nested;
+    typedef CGAL::Lazy_exact_nt<ET> Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/Lazy_exact_nt.h
+++ b/Number_types/include/CGAL/Lazy_exact_nt.h
@@ -1422,9 +1422,7 @@ class Modular_traits<Lazy_exact_nt<ET> >
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<typename ET> struct NumTraits<CGAL::Lazy_exact_nt<ET> >
-   : GenericNumTraits<CGAL::Lazy_exact_nt<ET> >
   {
     typedef CGAL::Lazy_exact_nt<ET> Real;
     // typedef CGAL::Lazy_exact_nt<ET> NonInteger;

--- a/Number_types/include/CGAL/Lazy_exact_nt.h
+++ b/Number_types/include/CGAL/Lazy_exact_nt.h
@@ -1422,7 +1422,9 @@ class Modular_traits<Lazy_exact_nt<ET> >
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<typename ET> struct NumTraits<CGAL::Lazy_exact_nt<ET> >
+   : GenericNumTraits<CGAL::Lazy_exact_nt<ET> >
   {
     typedef CGAL::Lazy_exact_nt<ET> Real;
     // typedef CGAL::Lazy_exact_nt<ET> NonInteger;

--- a/Number_types/include/CGAL/MP_Float.h
+++ b/Number_types/include/CGAL/MP_Float.h
@@ -888,6 +888,7 @@ namespace Eigen {
     typedef CGAL::MP_Float Real;
     typedef CGAL::Quotient<CGAL::MP_Float> NonInteger;
     typedef CGAL::MP_Float Nested;
+    typedef CGAL::MP_Float Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/MP_Float.h
+++ b/Number_types/include/CGAL/MP_Float.h
@@ -883,7 +883,9 @@ CGAL_DEFINE_COERCION_TRAITS_FROM_TO(int, MP_Float)
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::MP_Float>
+   : GenericNumTraits<CGAL::MP_Float>
   {
     typedef CGAL::MP_Float Real;
     typedef CGAL::Quotient<CGAL::MP_Float> NonInteger;

--- a/Number_types/include/CGAL/MP_Float.h
+++ b/Number_types/include/CGAL/MP_Float.h
@@ -883,9 +883,7 @@ CGAL_DEFINE_COERCION_TRAITS_FROM_TO(int, MP_Float)
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::MP_Float>
-   : GenericNumTraits<CGAL::MP_Float>
   {
     typedef CGAL::MP_Float Real;
     typedef CGAL::Quotient<CGAL::MP_Float> NonInteger;

--- a/Number_types/include/CGAL/Mpzf.h
+++ b/Number_types/include/CGAL/Mpzf.h
@@ -1135,6 +1135,7 @@ namespace Eigen {
     /* Should this be Quotient<Mpzf>? Gmpq?  */
     typedef CGAL::Mpzf NonInteger;
     typedef CGAL::Mpzf Nested;
+    typedef CGAL::Mpzf Literal;
 
     static inline Real epsilon() { return 0; }
     static inline Real dummy_precision() { return 0; }

--- a/Number_types/include/CGAL/Mpzf.h
+++ b/Number_types/include/CGAL/Mpzf.h
@@ -1129,7 +1129,9 @@ CGAL_DEFINE_COERCION_TRAITS_FROM_TO(mpz_class,Mpzf)
  */
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Mpzf>
+   : GenericNumTraits<CGAL::Mpzf>
   {
     typedef CGAL::Mpzf Real;
     /* Should this be Quotient<Mpzf>? Gmpq?  */

--- a/Number_types/include/CGAL/Mpzf.h
+++ b/Number_types/include/CGAL/Mpzf.h
@@ -1129,9 +1129,7 @@ CGAL_DEFINE_COERCION_TRAITS_FROM_TO(mpz_class,Mpzf)
  */
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<> struct NumTraits<CGAL::Mpzf>
-   : GenericNumTraits<CGAL::Mpzf>
   {
     typedef CGAL::Mpzf Real;
     /* Should this be Quotient<Mpzf>? Gmpq?  */

--- a/Number_types/include/CGAL/Quotient.h
+++ b/Number_types/include/CGAL/Quotient.h
@@ -859,6 +859,7 @@ namespace Eigen {
     typedef CGAL::Quotient<NT> Real;
     typedef CGAL::Quotient<NT> NonInteger;
     typedef CGAL::Quotient<NT> Nested;
+    typedef CGAL::Quotient<NT> Literal;
 
     static inline Real epsilon() { return NumTraits<NT>::epsilon(); }
     static inline Real dummy_precision() { return NumTraits<NT>::dummy_precision(); }

--- a/Number_types/include/CGAL/Quotient.h
+++ b/Number_types/include/CGAL/Quotient.h
@@ -854,7 +854,9 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template<class NT> struct NumTraits<CGAL::Quotient<NT> >
+   : GenericNumTraits<CGAL::Quotient<NT> >
   {
     typedef CGAL::Quotient<NT> Real;
     typedef CGAL::Quotient<NT> NonInteger;

--- a/Number_types/include/CGAL/Quotient.h
+++ b/Number_types/include/CGAL/Quotient.h
@@ -854,9 +854,7 @@ public:
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template<class NT> struct NumTraits<CGAL::Quotient<NT> >
-   : GenericNumTraits<CGAL::Quotient<NT> >
   {
     typedef CGAL::Quotient<NT> Real;
     typedef CGAL::Quotient<NT> NonInteger;

--- a/Number_types/include/CGAL/Sqrt_extension/Eigen_NumTraits.h
+++ b/Number_types/include/CGAL/Sqrt_extension/Eigen_NumTraits.h
@@ -30,6 +30,7 @@ namespace Eigen {
     typedef CGAL::Sqrt_extension<NT, ROOT, ACDE_TAG, FP_TAG> Real;
     typedef Real NonInteger;
     typedef Real Nested;
+    typedef Real Literal;
 
     static inline Real epsilon() { return NumTraits<NT>::epsilon(); }
 

--- a/Number_types/include/CGAL/Sqrt_extension/Eigen_NumTraits.h
+++ b/Number_types/include/CGAL/Sqrt_extension/Eigen_NumTraits.h
@@ -24,10 +24,8 @@
 
 namespace Eigen {
   template<class> struct NumTraits;
-  template<class> struct GenericNumTraits;
   template <class NT,class ROOT, class ACDE_TAG, class FP_TAG>
   struct NumTraits<CGAL::Sqrt_extension<NT, ROOT, ACDE_TAG, FP_TAG> >
-   : GenericNumTraits<CGAL::Sqrt_extension<NT, ROOT, ACDE_TAG, FP_TAG> >
   {
     typedef CGAL::Sqrt_extension<NT, ROOT, ACDE_TAG, FP_TAG> Real;
     typedef Real NonInteger;

--- a/Number_types/include/CGAL/Sqrt_extension/Eigen_NumTraits.h
+++ b/Number_types/include/CGAL/Sqrt_extension/Eigen_NumTraits.h
@@ -24,8 +24,10 @@
 
 namespace Eigen {
   template<class> struct NumTraits;
+  template<class> struct GenericNumTraits;
   template <class NT,class ROOT, class ACDE_TAG, class FP_TAG>
   struct NumTraits<CGAL::Sqrt_extension<NT, ROOT, ACDE_TAG, FP_TAG> >
+   : GenericNumTraits<CGAL::Sqrt_extension<NT, ROOT, ACDE_TAG, FP_TAG> >
   {
     typedef CGAL::Sqrt_extension<NT, ROOT, ACDE_TAG, FP_TAG> Real;
     typedef Real NonInteger;

